### PR TITLE
docs: cleanup wasm_hook.go

### PR DESCRIPTION
**Pull request by robert-zaremba**
Updated few comments and simplified the code.

### CODE REVIEW

File: modules/ibc-hooks/wasm_hook.go [hunk 1]
```diff
@@ -181,7 +177,6 @@ func ValidateAndParseMemo(memo string, receiver string) (isWasmRouted bool, cont
 			fmt.Errorf(types.ErrBadMetadataFormatMsg, memo, "wasm metadata is not a valid JSON map object")
 	}
 
-	// Get the contract
```
Comment (robert-zaremba) [inline modules/ibc-hooks/wasm_hook.go:184]: this comment doesn't bring anything. The implementation and the error message is clear and short, so it's better to read the code than the comment.
[Rule 1: Comments should not duplicate the code](https://stackoverflow.blog/2021/12/23/best-practices-for-writing-code-comments)

Comment (robert-zaremba) [reply modules/ibc-hooks/wasm_hook.go:184]: same comment applies to few changes below

**Comment by Reecepbcups** [comment]
The team just updated main to SDK v50, if you want to rebase on that I am happy to approve myself to get in the updated docs

**Review by Reecepbcups — APPROVED** [review]
approving as it is just minor formatting & in code documentation after significant wait.

**Pull request merged**